magpie/ui/network/views.py

```diff
@@ -0,0 +1,45 @@
+import jwt
+from pyramid.authentication import Authenticated
+from pyramid.httpexceptions import HTTPBadRequest
+from pyramid.view import view_config
+
+from magpie.api.management.network.network_utils import decode_jwt, encode_jwt
+from magpie.models import NetworkNode
+from magpie.ui.utils import BaseViews
+from magpie.utils import get_logger
+
+LOGGER = get_logger(__name__)
+
+
+class NetworkViews(BaseViews):
+    @view_config(route_name="authorize", renderer="templates/authorize.mako", permission=Authenticated)
+    def authorize(self):
+        token = self.request.GET.get("token")
+        response_type = self.request.GET.get("response_type")
+        redirect_uri = self.request.GET.get("redirect_uri")
+
+        # Extend this to other response types later if needed
+        if response_type != "id_token":
+            raise HTTPBadRequest("Invalid response type")
+        if token is None:
+            raise HTTPBadRequest("Missing token")
+        try:
+            node_name = jwt.decode(token, options={"verify_signature": False}).get("iss")
```

Comment:
I think the JWT checks and encoding strategies are too specific to be done on the UI side. 
They should all be done within the API to avoid duplicating this logic, and have a way to do the same directly with API requests.

We should limit quick pre-checks only to obvious issues such as the bad response type or empty token above to avoid refreshing the whole UI page only for an error, which will cause more requests/processing than simply returning the error here.

Reply:
Alright, I'll try to reorganize a bit and see how it looks